src/pages/SettingsPage/UsersSettings/UserList.jsx

```diff
@@ -29,10 +29,21 @@ const UserList = ({
 }) => {
   // Selection
   const selection = useMemo(
-    () => userList.filter((user) => selectedUsers.includes(user.name)),
+    () => {
+      return userList.filter((user) => selectedUsers.includes(user.name))
+    },
     [selectedUsers, selectedProjects, userList],
   )
 
+  const onContextMenu = (e) => {
+    if (selectedUsers.includes(e.data.name)) {
+      onSelectUsers([e.data.name, ...selectedUsers])
+    } else {
+      onSelectUsers([e.data.name])
+    }
+    ctxMenuShow(e.originalEvent, ctxMenuItems(selectedUsers))
```

Comment:
The new selection can either be the current `selectedUsers` **or** `[e.data.name]`. This only ever passes `selectedUsers` which could be wrong if `onSelectUsers([e.data.name])` is called.

Use a new constant `newSelectedUsers` that gets defined above based on the if statement.